src/main/software/amazon/event/ruler/Patterns.java

```diff
@@ -85,12 +86,22 @@ public static AnythingBut anythingButNumberMatch(final Set<Double> anythingButs)
         return new AnythingBut(normalizedNumbers, true);
     }
 
-    public static ValuePatterns anythingButPrefix(final String prefix) {
-        return new ValuePatterns(MatchType.ANYTHING_BUT_PREFIX, prefix);
+    public static AnythingButValuesSet anythingButPrefix(final String prefix) {
+        return new AnythingButValuesSet(MatchType.ANYTHING_BUT_PREFIX, Collections.singleton(prefix));
     }
 
-    public static ValuePatterns anythingButSuffix(final String suffix) {
-        return new ValuePatterns(MatchType.ANYTHING_BUT_SUFFIX, new StringBuilder(suffix).reverse().toString());
+    public static AnythingButValuesSet anythingButPrefix(final Set<String> anythingButs) {
+        return new AnythingButValuesSet(MatchType.ANYTHING_BUT_PREFIX, anythingButs);
+    }
+
+    public static AnythingButValuesSet anythingButSuffix(final String suffix) {
+        return new AnythingButValuesSet(MatchType.ANYTHING_BUT_SUFFIX,
+                Collections.singleton(new StringBuilder(suffix).reverse().toString()));
+    }
+
+    public static AnythingButValuesSet anythingButSuffix(final Set<String> anythingButs) {
+        return new AnythingButValuesSet(MatchType.ANYTHING_BUT_SUFFIX,
+                anythingButs.stream().map(s -> new StringBuilder(s).reverse().toString()).collect(Collectors.toSet()));
```

Comment:
Do these need to be mentioned in README ? in the patterns API section?

Reply:
That section was quite out of date. I've brought it up to date.